OpenRA.Game/FileSystem/InstallShieldCABExtractor.cs

```diff
@@ -404,9 +405,44 @@ public string DirectoryName(uint index)
 			return test;
 		}
 
+		public string GetFirstSubDirectory(string package)
+		{
+			var dir = "";
+			var delimeter = "{0}".F(Path.DirectorySeparatorChar);
+
+			foreach (var key in fileLookup.Keys)
+				if (key.StartsWith(package.ToLowerInvariant()) && key.Contains(delimeter))
+				{
+					var tmp = key.Split(Path.DirectorySeparatorChar)[1];
+
+					if (!tmp.Contains("."))
+					{
+						if (tmp == package)
+							tmp = "";
+
+						dir = tmp;
+						break;
+					}
+					else
+						break;
+				}
+
+			return dir;
+		}
+
+		public uint GetIndexByFilename(string directory, string filename)
+		{
+			var path = directory + Path.DirectorySeparatorChar + filename;
+
+			if (Exists(path.ToLowerInvariant()))
+				return fileLookup[path.ToLowerInvariant()];
+			else
```

Comment:
that else handles conditions like when files can not be found in the file table... it must return indexe i can not return 0 because zero stays for a another file in the lookup table